src/storage/combiner.jl

```diff
@@ -0,0 +1,89 @@
+export Combiner
+
+struct CombinerStorage <: TensorStorage
+    ci::Index
+    inds::IndexSet
```

Comment:
Did the code end up using `ci` and `inds` stored inside CombinerStorage? If not, perhaps CombinerStorage can just hold no data.

Reply:
It doesn't yet but it might be nice to have the Combiner carry around which index is the combined one

Reply:
True, but let’s make sure we have a concrete application of this choice. I think a good one would be a function `combinedIndex` which acts on an ITensor with combiner storage, and returns `ci`. We have that function in C++ but the way it works relies on the first index being the combined one, which is a brittle way to do it. 

So how about we keep `ci` but remove `inds`. We can always add `inds` back in later when it is needed without breaking any existing code.

Reply:
Sure!